allennlp/data/dataset_readers/reading_comprehension/squad.py

```diff
@@ -26,6 +26,11 @@ class SquadReader(DatasetReader):
     ``metadata['token_offsets']``.  This is so that we can more easily use the official SQuAD
     evaluation script to get metrics.
 
+    We also support limiting the maximum passage length and question length for training and evaluation.
+    Since some answer spans may exceed the maximum passage length, we regard such answer as invalid.
+    During training, we skip those examples of which all answer spans are invalid (invalid examples);
+    During testing, we use the last token as the pseudo gold answer for these invalid examples.
```

Comment:
I'm trying to understand what happens here.  We still have the correct answer string for the evaluation script, so we'll get it wrong no matter what we predict?  Is that right?  That's what should happen, so if that's what's happening, great, and we should make it more clear in the documentation somehow.

Reply:
Yes, that's right. The correct answer string does not change. The span should be skipped, or replaced with a pseudo span if it exceeds the cut passage length. Otherwise, it will cause error in making the IndexField for span.

Reply:
Great, can you update the docstring to mention this.  Even better, it'd be great to have a test showing that this still works as intended - we _really_ don't want to compute inaccurate metrics at validation / test time.